cmake/kokkos_enable_options.cmake

```diff
@@ -160,6 +161,16 @@ IF(Kokkos_ENABLE_CUDA_LDG_INTRINSIC)
     MESSAGE(FATAL_ERROR "Kokkos_ENABLE_CUDA_LDG_INTRINSIC has been removed. LDG intrinsics are always enabled.")
   ENDIF()
 ENDIF()
+IF(NOT Kokkos_ENABLE_CUDA_LAMBDA)
```

Comment:
Meaning we do not warn when users enable.  I propose we keep it this way for a number of releases and get rid of it closer to 5.0

Reply:
I'm not against this.

However, why wait to warn?  Is it because we expect people to switch between Kokkos 3 and 4 for a while, and we don't want to add friction?